File: frontend/src/components/Transaction/Transaction.js
// Transaction.js
const Transaction = ({ transaction }) => {
    return (
      <div className="transaction">
        <p>Transaction ID: {transaction.transactionId}</p>
        <p>User ID: {transaction.userId}</p>
        <p>Symbol: {transaction.symbol}</p>
        <p>Shares: {transaction.shares}</p>
        <p>Transaction Type: {transaction.transactionType}</p>
        <p>Price: {transaction.price}</p>
        <p>Total Value: {transaction.totalValue}</p>
        <p>Timestamp: {new Date(transaction.timestamp).toLocaleString()}</p>
      </div>
    );
  };
  
  export default Transaction;
  

File: scorecalculation-microservice/src/main/java/com/trade/scorecalculation_microservice/dto/Transaction.java
package com.trade.scorecalculation_microservice.dto;

import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "transaction")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    @Id
    private String transactionId;
    private String userId;
    
    private String symbol;
    // Enum: BUY / SELL
    private String transactionType;

    private Double shares;
    private Double price;
    private Double totalValue;

    // private Double portfolioValue;
    // private Double pnlRealized;

    private LocalDateTime timestamp;
}
